src/main/java/com/blog/web/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.blog.web.services.impl;
+
+import com.blog.web.dto.RegistrationDto;
+import com.blog.web.models.Role;
+import com.blog.web.models.UserEntity;
+import com.blog.web.repository.RoleRepository;
+import com.blog.web.repository.UserRepository;
+import com.blog.web.services.UserService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private UserRepository userRepository;
```

Comment:
make variables final to ensure immutability -> calling something final should be the default, having something not final means coder needs to see if it gets mutated